ui/pages/keychains/restore-vault.js

```diff
@@ -44,13 +44,14 @@ class RestoreVaultPage extends Component {
     const { t } = this.context;
     let seedPhraseError = null;
 
-    const wordCount = this.parseSeedPhrase(seedPhrase).split(/\s/u).length;
+    const parseSeedPhrase = this.parseSeedPhrase(seedPhrase);
+    const wordCount = parseSeedPhrase.split(/\s/u).length;
     if (
-      seedPhrase &&
+      parseSeedPhrase &&
       (wordCount % 3 !== 0 || wordCount < 12 || wordCount > 24)
     ) {
       seedPhraseError = t('seedPhraseReq');
-    } else if (!isValidMnemonic(seedPhrase)) {
+    } else if (!isValidMnemonic(parseSeedPhrase)) {
```

Comment:
It seems like we're now validating the "parsed seed phrase" rather than the actual seed phrase that gets used for the vault restoration. This might result in the validation passing here, but failing when the vault gets restored. This seems bad.

Reply:
https://github.com/MetaMask/metamask-extension/blob/401a081445834c9bb255509501238b04869d9bef/ui/pages/keychains/restore-vault.js#L100

Reply:
Right, just discovered that myself too! Thanks.